overlord/snapstate/snapstate.go

```diff
@@ -1881,12 +1885,32 @@ type update struct {
 // revision of the snap.
```

Comment:
it seems the doc comment for this needs changes, also I wonder if turning this around and calling it revisionDiverged  wouldn't be clear?

Reply:
I personally think that "satisfied" works better here as a method on `update`, since it reads as the update has already been satisfied. Using diverged implies something different to me, almost as if something has unexpectedly changed.